content/tutorial-regular-keys.md

```diff
@@ -0,0 +1,274 @@
+2. [Assign the keys to your account as regular keys](#2-assign-keys-to-your-account-as-regular-keys)
+3. [Submit a test payment using regular keys](#3-submit-a-test-payment-using-regular-keys)
+4. [Explore next steps](#4-explore-next-steps)
+
+
+### 1. Generate Keys
+
+Use the [`wallet_propose`](reference-rippled.html#wallet-propose) method to generate the keys that you'll assign to your account as regular keys.
+
+Here's an example `wallet_propose` request and response:
+
+
+#### WebSocket Request
+
+```
+{
+  "method": "wallet_propose"
+}
+```
+
+#### WebSocket Response
+
+```
+{
+ "result": {
+   "account_id": "rf9vUoqa83foreMr6CYFCEPTZUFagZv83c",
+   "key_type": "secp256k1",
+   "master_key": "REG ULAR PRIV ATE KEY RFC 1751 XXX XXXX XXX XXXX XXX",
+   "master_seed": "RegularPrivateKeyXXXXXXXXXXXX",
+   "master_seed_hex": "RegularPrivateKeyHexXXXXXXXXXXXX",
+   "public_key": "aBR9dmoHfrfU24q1cbjdehNsFiuV9k5PrrQTLiH42qxhWr5ybc1K",
+   "public_key_hex": "03A45E8E4BFECA7473F932839B982C33CD2FB7B788932A21DCDD8536AE63A1614C"
+ },
+ "status": "success",
+ "type": "response"
+}
+```
+
+In the next step, you'll use the `account_id` from this response to assign the keys as regular keys to your account.
+
+
+### 2. Assign Keys to Your Account as Regular Keys
+
+Use the [`SetRegularKey`](reference-transaction-format.html#setregularkey) method to assign the keys you generated in step 1 to your account as regular keys.
+
+In the request, send your account's `account_ID` as the `account` value, your account's `master_key`, `master_seed`, or `master_seed_hex` (master private key) as the `secret` value, and the `account_id` generated in step 1 as the `RegularKey` value.
+
+Here's an example `SetRegularKey` request and response.
+
+
+#### WebSocket Request:
+
+**Note:** This request format uses the `submit` command and a `tx_json` object to sign and submit this transaction all at once. This practice is recommended for **testing and development purposes only**. Transmitting your account secret is dangerous, so you should do this only from within a trusted and encrypted connection, or through a local connection, and only to a server you control.
+
+***TODO: Question: Should this tutorial walk folks through the process of offline signing and then submission? It feels weird to say that they should keep their master keys offline and then have them send their secret in the request below. Is our approach to tutorials that they should show a user how something works for test and dev purposes? Just to show results quickly? Or is it to illustrate best practices for how a task should be performed day-to-day? If yes, then I think I need to get ripple-lib setup to do offline signing and a submit here in the tutorial. https://ripple.com/build/rippleapi/#offline-functionality -- this offline signing is not working for me right off the bat. Errors are not being returned, so I can't troubleshoot. I am sure that I'm missing a few steps -- will require research.***
```

Comment:
Thank you for such a thorough and thoughtful reply, Rome. I get it. I think you are right that is a tutorial that needs to be in the “do it right” category -- if only because it will contradict its fundamental premise if it doesn’t. The only hesitation I have is that to be able to write this tutorial the “do it right” way, I’ll need to write an offline signing tutorial that will feed into this tutorial first. I can and will do it, happily -- it’ll just hold up publication of this tutorial. @mDuo13 - what do you think of publishing this tutorial without the offline signing distinction -- but calling out that sending the secret in a sign and submit transaction is not a best practice for every day production use -- but is used in this tutorial to provide a quick demonstration of functionality for test and dev purposes only? And then when I get the offline signing tutorial working, we can update the tutorial to provide that path for folks who want to understand and use the production best practices flow?

Reply:
That sounds pretty good, but let's take it a step closer to offline signing by using the `sign` command separately and then submitting the signed blob using the `submit` command with the generated `tx_blob` instead.

Then you can have a little verbiage about how the "sign" command can be done on a separate machine or with a signing library and the rest should more or less fit into the paradigm for proper offline signing.